opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/Open.java

```diff
@@ -402,11 +406,22 @@ private String getCoverUrl(Element img) {
         // caller=vlbPublic&amp;func=DirectIsbnSearch&amp;isbn=9783868511291&amp;
         // nSiteId=11|c|SetNoCover|a|/DesktopModules/OCLC.OPEN.PL.DNN
         // .BaseLibrary/StyleSheets/Images/Fallbacks/emptyURL.gif?4.2.0.0|a|
-        String url = null;
-        if (parts.length >= 2 && parts[0].equals("SetSimpleCover")) {
-            url = parts[2].replace("&amp;", "&");
+        for (int i = 0; i + 2 < parts.length; i++) {
+            if (parts[i].equals("SetSimpleCover")) {
+                String url = parts[i + 2].replace("&amp;", "&");
+                try {
+                    HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
+                    conn.setRequestMethod("HEAD");
```

Comment:
While I'm gonna merge this for now, I really don't like this pattern of doing a lot of HEAD calls inside search(). We have something similar for another API, IIRC. This makes the search() call a lot slower and we *still* have to download the file if we want to display the cover. I think we need either put a concept of "alternative cover URLs" into the API so the frontend can do this asynchronously or download the file here completely and include the bitmap in the SearchResult.